### PR TITLE
Fix free/delete inconsistencies in OSystem_Win32::convertEncoding

### PR DESCRIPTION
**Pull request by criezy**
The current implementation of OSystem_Win32::convertEncoding causes some arrays allocated with `new []` being freed using `free()` and not `delete []`. This is an attempt to fix this.

Since I am not even able to compile the code, not being on Windows, I would appreciate if others could review and try this before this is pushed to master.

This commit also removes what appeared to be an unused variable (`wString`) and adds a sanity check as the result of `Win32::ansiToUnicode` could be null.

### CODE REVIEW

**Comment by sluicebox** [comment]
I stepped through the changes in both codepaths on Windows with several test inputs and they are both fixed and doing what's intended.

The comment "Win32::unicodeToAnsi ses new to allocate" is missing the "u" and the project's code conventions are for no space in "delete []", which I only know from having just done this =)

File: backends/platform/sdl/win32/win32.cpp [hunk 1]
```diff
@@ -492,7 +498,15 @@ char *OSystem_Win32::convertEncoding(const char* to, const char *from, const cha
 	} else {
 		result = Win32::unicodeToAnsi(tmpStr, Win32::getCodePageId(to));
 		free(tmpStr);
-		return result;
+		if (!result)
+			return nullptr;
+		// Win32::unicodeToAnsi ses new to allocate the memory. We need to copy it into an array
```
Comment (SupSuper) [inline backends/platform/sdl/win32/win32.cpp]: Typo

**Comment by SupSuper** [comment]
Compiles fine here. Shame we can't just wrap them in managed Strings and avoid manual allocation, since they're Win32 types.

**Comment by criezy** [comment]
Thank you to both of you. I have fixed the typo and `delete[]` and I will now merge this.

**Pull request merged**